### ISSUE TITLE
Rescued soldier joins their original team that is on another mission

### ISSUE THREAD

**Issue by stann1** [long-war-2/lwotc#1576]
I did a a mission where I had to rescue a VIP, who was actually a soldier I had previously assigned as Haven adviser, but lost due to unchecked intel retaliation mission.

At the same time, this soldier's original squad was infiltrating another mission (4 days elapsed). When I rescued him and he rejoined my barracks, he was put immediately on the mission of his original team. Furthermore, the mission expiration was reset to its original state - 7days, but my infiltration percentage remained the same. 

So basically a huge boost for the player, but it is still a bug.
